controllers/progressiverollout_controller_test.go

```diff
@@ -315,8 +317,22 @@ var _ = Describe("ProgressiveRollout Controller", func() {
 			}
 			Expect(k8sClient.Create(ctx, singleStagePR)).To(Succeed())
 
+			createdPR := deploymentskyscannernetv1alpha1.ProgressiveRollout{}
+			Eventually(func() int {
+				_ = k8sClient.Get(ctx, client.ObjectKeyFromObject(singleStagePR), &createdPR)
+				return len(createdPR.ObjectMeta.Finalizers)
+			}).Should(Equal(1))
+			Expect(createdPR.ObjectMeta.Finalizers[0]).To(Equal(deploymentskyscannernetv1alpha1.ProgressiveRolloutFinalizer))
+
 			expected := singleStagePR.NewStatusCondition(deploymentskyscannernetv1alpha1.CompletedCondition, metav1.ConditionTrue, deploymentskyscannernetv1alpha1.StagesCompleteReason, "All stages completed")
 			ExpectCondition(singleStagePR, expected.Type).Should(HaveStatus(expected.Status, expected.Reason, expected.Message))
+
+			deletedPR := deploymentskyscannernetv1alpha1.ProgressiveRollout{}
+			Expect(k8sClient.Delete(ctx, singleStagePR)).To(Succeed())
+			Eventually(func() error {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(singleStagePR), &deletedPR)
+				return err
+			}).Should(HaveOccurred())
```

Comment:
the error occurs because we try to fetch an object that's been deleted, right? is the deletion supposed to be instantaneous or could the test be flaky here?
Would the Eventually be retried until the object is deleted?

Reply:
The idea behind this test is we delete the object, but since we need to remove the finalizer, it is not immediate. This is why we use eventually: we are able to get the object until it gets deleted, then we get an error - capture by the `Should(HaveOccurred())` clause